src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,229 @@
+        Diagnostics.Module_0_declares_1_locally_but_it_is_not_exported.code,
+    ];
+
+    registerCodeFix({
+        errorCodes,
+
+        getCodeActions(context) {
+            const { sourceFile } = context;
+
+            const info = getInfo(sourceFile, context, context.span.start);
+
+            if (!info || info.originSourceFile.isDeclarationFile) {
+                return undefined;
+            }
+
+            const changes = textChanges.ChangeTracker.with(context, (t) =>
+                doChange(t, info.originSourceFile, info.node)
+            );
+
+            return [
+                createCodeFixAction(
+                    /*fixName*/ fixId,
+                    changes,
+                    /*description*/ [
+                        Diagnostics.Export_0_from_module_1,
+                        info.node.text,
+                        showModuleSpecifier(info.importDecl),
+                    ],
+                    fixId,
+                    /*fixAllDescription*/ Diagnostics.Add_all_missing_exports
+                ),
+            ];
+        },
+
+        fixIds: [fixId],
+
+        getAllCodeActions: (context) =>
+            codeFixAll(context, errorCodes, (changes, diag) => {
+                const info = getInfo(diag.file, context, diag.start);
+
+                if (info) {
+                    doChange(changes, info.originSourceFile, info.node);
+                }
+            }),
+    });
+
+    interface Info {
+        readonly node: Identifier;
+
+        readonly importDecl: ImportDeclaration;
+
+        readonly originSourceFile: SourceFile;
+    }
+
+    function getInfo(
+        sourceFile: SourceFile,
+        context: CodeFixContext | CodeFixAllContext,
+        pos: number
+    ): Info | undefined {
+        const node = getTokenAtPosition(sourceFile, pos);
+
+        if (!isIdentifier(node)) {
+            return;
+        }
+
+        const importDecl = findAncestor(node, isImportDeclaration);
+
+        if (!importDecl || !isStringLiteralLike(importDecl.moduleSpecifier)) {
+            return undefined;
+        }
+
+        const resolvedModule = getResolvedModule(
+            sourceFile,
+            /*moduleName*/ importDecl.moduleSpecifier.text,
+            /*mode*/ undefined
+        );
+
+        if (!resolvedModule) {
+            return undefined;
+        }
+
+        const originSourceFile = context.program.getSourceFile(
+            resolvedModule.resolvedFileName
+        );
+
+        if (!originSourceFile) {
+            return undefined;
+        }
+
+        return { node, importDecl, originSourceFile };
+    }
+
+    function getNamedExportDeclaration(
+        sourceFile: SourceFile
+    ): ExportDeclaration | undefined {
+        let namedExport;
+
+        for (const statement of sourceFile.statements) {
+            if (
+                isExportDeclaration(statement) &&
+                statement.exportClause &&
+                isNamedExports(statement.exportClause)

```

Comment:
We should skip re-export statements here: `export { a } from "./something";`.

Reply:
yup nice, added